src/main/java/org/apache/sling/jcr/resource/internal/helper/jcr/JcrResourceProvider.java

```diff
@@ -357,19 +356,17 @@ public Resource getResource(ResolveContext<JcrProviderState> ctx, String path, R
                     version = child.getResourceMetadata().getParameterMap().get("v");
                 }
                 if (version == null) {
-                    Item item = ((JcrItemResource<?>) child).getItem();
-                    if ("/".equals(item.getPath())) {
-                        return null;
-                    }
-                    Node parentNode;
-                    try {
-                        parentNode = item.getParent();
-                    } catch(AccessDeniedException e) {
-                        return null;
-                    }
                     String parentPath = ResourceUtil.getParent(child.getPath());
-                    return new JcrNodeResource(ctx.getResourceResolver(), parentPath, version, parentNode,
-                            ctx.getProviderState().getHelperData());
+                    if (parentPath != null) {
+                        Node parentNode = (Node) ctx.getProviderState().getResourceFactory()
+                            .getItemOrNull(parentPath);
+                        if (parentNode != null) {
+                            return new JcrNodeResource(ctx.getResourceResolver(),
+                                parentNode.getPath(), null, parentNode,
+                                ctx.getProviderState().getHelperData());
+                        }
```

Comment:
You could get rid of the unchecked cast quite easily:

    Item parentItem = ctx.getProviderState().getResourceFactory()
        .getItemOrNull(parentPath);
    if (parentItem instanceof Node) {
        return new JcrNodeResource(ctx.getResourceResolver(),
            parentPath, null, (Node)parentItem,
            ctx.getProviderState().getHelperData());
        }
    }

Reply:
parentItem.isNode() :)

Reply:
fair point :)